src/Package.js

```diff
@@ -96,6 +96,9 @@ export default class Package {
     // check if semantic versions are compatible
     if (semver.satisfies(actualVersion, expectedVersion)) {
       return true;
+    // or the dependency is specified as a relative folder
+    } else if (/^file:/.test(dependency.version)) {
```

Comment:
i think `file:` is not actually necessary, one can do:
```json
{
  "dependencies": {
    "lodash": "../../node_modules/lodash"
  }
}
```

I would check npm repo how they test whether a version is a local path.